### PR TITLE
dp: pt: rearrange folder structure of atomic_model and model.

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
- create models `dpmodel.atomic_model` and `pt.model.atomic_model`. 
- place all the atomic model related codes into the new modules. 
- `PairTabModel` -> `PairTabAtomicModel`
- `ZBLModel` -> `DPZBLModel`

### CODE REVIEW

File: deepmd/dpmodel/atomic_model/__init__.py
Comment (njzjz) [inline deepmd/dpmodel/atomic_model/__init__.py:1]: Module-level docstring is required, so the developer can distinguish `atomic_model` from `model`.

File: deepmd/dpmodel/model/__init__.py
Comment (njzjz) [inline deepmd/dpmodel/model/__init__.py:1]: Module-level docstring is required.

File: deepmd/pt/model/atomic_model/pairtab_atomic_model.py [hunk 1]
```diff
@@ -19,12 +19,9 @@
 from .base_atomic_model import (
     BaseAtomicModel,
 )
-from .model import (
-    BaseModel,
-)
 
 
-class PairTabModel(BaseModel, BaseAtomicModel):
+class PairTabAtomicModel(torch.nn.Module, BaseAtomicModel):
```
Comment (anyangml) [inline deepmd/pt/model/atomic_model/pairtab_atomic_model.py:24]: I changed this to BaseModel in the CUDA PR, I think this is fine, but we should probably test it to be sure.

Comment (wanghan-iapcm) [reply deepmd/pt/model/atomic_model/pairtab_atomic_model.py:24]: I find the current `BaseModel` (after #3245) provides nothing and can be safely removed.

Comment (wanghan-iapcm) [reply deepmd/pt/model/atomic_model/pairtab_atomic_model.py:24]: By name `BaseModel` seems to be the common base class for all models but not for all atomic models.

**Comment by anyangml** [comment]
LGTM

**Review by anyangml — APPROVED** [review]

**Pull request merged**